.github/workflows/deploy.yml

```diff
@@ -5,9 +5,13 @@ on:
     branches:
       - main
 
+permissions:
+  contents: write
+
 jobs:
   deploy:
     runs-on: ubuntu-latest
+    environment: production
```

Comment:
Where this env is defined?

Reply:
In the repo settings. NPM_TOKEN is a env-protected secret that available only in `production` env and `main` branch